File: queueWithFIFO.py
class Node:
    def __init__(self, d):
        self.data = d
        self.next = None

class Queue:
    def __init__(self):
        self.front = None
        self.rear = None

    def enqueue(self, d):
        newnode = Node(d)
        newnode.next = self.front
        self.front = newnode
        if self.rear is None:
            self.rear = self.front

    def dequeue(self):
        if self.rear is None:
            return 0
        if self.rear == self.front:
            a = self.rear.data
            self.rear = None
            self.front = None
            return a
        first = self.front
        while first.next != self.rear:
            first = first.next
        a = self.rear.data
        first.next = None
        self.rear = first
        return a

    def display(self):
        if self.front is None:
            return
        temp = self.front
        while temp is not None:
            print(temp.data)
            temp = temp.next

if __name__ == "__main__":
    q = Queue()
    q.enqueue(1)
    q.enqueue(2)
    q.enqueue(3)
    q.enqueue(4)
    q.display()
    print(13456)
    q.dequeue()
    q.display()

File: QueueWithFIFO.java
public class QueueWithFIFO {
    public static void main(String[] args) {
        Queue  q=new Queue();
        q.enqueue(1);
        q.enqueue(2);
        q.enqueue(3);
        q.enqueue(4);
        q.display();
        System.out.println(13456);
        q.dequeue();
        q.display();

    }
}

class Node{
    int data;
    Node next;
    Node(int d){
        this.data=d;
        this.next=null;
    }
}

class Queue{
    Node front;
    Node rear;
    Queue(){
        this.front=null;
        this.rear=null;
    }

    public void enqueue(int d){
        Node newnode = new Node(d);
        newnode.next=this.front;
        this.front=newnode;
        if(this.rear==null){
            this.rear=front;
        }
       
    }

    public int dequeue(){
        if(rear==null) return 0;
        if(rear==front){
            int a = rear.data;
            rear=null;
            front=null;
            return a;
        }
        Node first = front;
        while(first.next!=rear){
            first=first.next;
        }
        first.next=null;
        int a=rear.data;
        rear=first;
    
        return a;
    }

    public void display(){
        if(this.front==null)return;
        Node temp=this.front;
        while(temp!=null){
            System.out.println(temp.data);
            temp=temp.next;
        }
    }
}
